tests/ui/pointer_in_nomem_asm_block.rs

```diff
@@ -0,0 +1,29 @@
+#![warn(clippy::pointer_in_nomem_asm_block)]
+#![crate_type = "lib"]
+#![no_std]
+
+use core::arch::asm;
+
+unsafe fn nomem_bad(p: &i32) {
+    asm!(
```

Comment:
Can we have error annotations here (and other cases where we want it to warn), like `//~^ ERROR: pointer_in_nomem_asm_block`?